OpenTrons/opentrons-api
# feat(components, protocol-designer): create Thermocycler form (excluding cycles)

# Overview

This PR adds the UI for the Thermocycler step form toolbox in PD redesign. It also minorly refactors several shared components implicated in the new Thermocycler form components. Note that cycle step creation will be wired up in a following PR.

Also of note, to match designs, I copied the `ToggleButton` component from app to PD. Do we think we move this component to the shared components library and replace all instances across app and PD?  

https://github.com/user-attachments/assets/65283058-8ce3-4d4e-b3b3-1e3352713071

AUTH-810

## Test Plan and Hands on Testing

- create or import a PD protocol with redesign flag on 
- create a new thermocycler step
- select 'change thermocycler state'
- select and input the toggle input and toggle form fields and ensure error messages show when appropriate
- create a new thermocycler step
- select 'program a thermocycler profile'
- select and input the toggle input and toggle form fields and ensure error messages show when appropriate
- select 'no profile defined' under 'protocol steps'
- verify that profile modal opens with "no profile steps" info screen
- click through adding, editing, and deleting steps, and verify that input fields validate with proper errors and masking
- save modal and verify that the number of steps correctly shows under "profile steps". clicking that button now opens the modal with steps populated

## Changelog

- create and wire up new components for Thermocycler step form
- add translations
- update `InputField` `StyledText` component from legacy to current
- correctly pass padding prop to `ListButton`
- fix title bottom border position in `Toolbox`
- add duplicate `ToggleButton` component to PD
- refactor title and caption usage in `InputStepFormField`
- swap out toggle. button in ToggleStepFormField and ToggleExpandStepFormField
- utilize `InputStepFormField` in `PauseTools`
- mask `profileVolume` form field to float
- check time errors for mm:ss format

## Review requests

See test plan. What do we think we should do about `ToggleButton`? 

## Risk assessment

low
